forbidigo/patterns.go

```diff
@@ -8,8 +8,9 @@ import (
 )
 
 type pattern struct {
-	pattern *regexp.Regexp
-	msg     string
+	pattern      *regexp.Regexp
+	msg          string
+	matchPackage bool
```

Comment:
I worry this name doesn't quite express the intent of this control (because it suggests to me we might only be matching the package name).  I can't think of a great name that isn't really long but maybe we can at least at leave a comment here about how this will be used.